projects/plugins/jetpack/extensions/blocks/ai-assistant/extensions/ai-assistant/with-ai-assistant.tsx

```diff
@@ -106,6 +112,13 @@ export const withAIAssistant = createHigherOrderComponent(
 		);
 
 		const replaceWithAiAssistantBlock = useCallback( () => {
+			const clientIds = select( 'core/block-editor' ).getSelectedBlockClientIds();
+			const blocks = select( 'core/block-editor' ).getBlocksByClientId( clientIds );
+			const content = blocks
+				.filter( block => block != null ) // Safeguard against null or undefined blocks
+				.map( block => getBlockTextContent( block.clientId ) )
+				.join( '\n\n' );
```

Comment:
Can we make this reusable to avoid duplication?

Reply:
yes, it deserves more love. I just wanted to share the code asap